### PR TITLE
DES 3.12 (new formula).

### PR DESCRIPTION
**Pull request by lifepillar**
DES (Datalog Educational System) is an educational Deductive database with Datalog, Relational Algebra and SQL.

I have written my own shim scripts, because DES must be executed from inside its own directory.

### CODE REVIEW

File: des.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class Des < Formula
+  desc "Datalog Educational System"
+  homepage "http://des.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/des/des/des3.12/DES3.12ACIDE0.17UnixesSWI.zip"
+  version "3.12"
+  sha256 "1cba61cfbe0d64e4ccf425691b650a352e43047bef4622262acd66b8eb3d62d5"
+
+  depends_on :java => "1.6+"
+  depends_on "swi-prolog"
```
Comment (dpo) [inline des.rb]: Should be `homebrew/x11/swi-prolog`.

File: des.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  def shim_script
+    <<-EOS.undent
```
Comment (dpo) [inline des.rb]: Please merge lines 11 and 12, as in `caveats`.

File: des.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      #!/usr/bin/env bash
+      cd #{libexec}
+      swipl des.pl
+    EOS
+  end
+
+  def jar_script(target_jar)
+    <<-EOS.undent
```
Comment (dpo) [inline des.rb]: Same here.

File: des.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        #{bin}/des
+
+    The executable to launch the GUI version of DES is
+        #{bin}/des-gui
+    EOS
+  end
+
+  test do
+    system "false"
+  end
```
Comment (dpo) [inline des.rb]: This won't do. Is there a useful test to run here? For example, the test could run `#{bin}/des` to output the version number.

Comment (lifepillar) [inline des.rb]: I have prepared a test. Is there a way to test it locally?

Comment (lifepillar) [inline des.rb]: Never mind, I've read about `brew test`.

**Pull request closed without merging**